report/types/report-result.d.ts

```diff
@@ -17,7 +17,8 @@ interface ReportResult extends LHResult {
 
 declare module ReportResult {
   interface Category extends LHResult.Category {
-    auditRefs: Array<AuditRef>
+    auditRefs: Array<AuditRef>;
+    displayMode: 'gauge'|'fraction';
```

Comment:
thinking of the API proposal in https://github.com/GoogleChrome/lighthouse/pull/12772/ (where `prepareReportResult` won't necessarily be used) and the possibility there might be more changes needed to category rendering besides just gauge vs fraction, what do you think about an options object passed into `renderCategoryScore`?

It seems like that's ultimately what we're going to want from `renderReport()`/`renderReportPartials()` and it would be nice to not alter `ReportResult` further from the LHR.

Reply:
> what do you think about an options object passed into renderCategoryScore?

That would mean passing the options through multiple layers of function calls, in multiple different locations. We could do it, but it seems like `ReportResult` was designed so we don't have to make these kinds of structural changes.

Reply:
`ReportResult` wasn't so much designed :)

https://github.com/GoogleChrome/lighthouse/blob/5cbad246937b6332aca9be65780c98a20bc8926d/report/renderer/util.js#L55

It is a structural change and it would have to go through a few layers (though not so many). I think it may be worth it, though:
- I want to call `renderGaugeForCategory(category, groups, options)`, I don't want to have to call `prepareReportResult` first
- if there are other per-mode changes (e.g. if we go with #13022, perf is going to be different than the other categories in navigation reports but not in timespan/snapshot), we're going to have to expand `displayMode` or generalize it anyways

basically if we're going to parameterize on gatherMode, let's just do it outright rather than roundabout through `ReportResult` :)